datafusion/sql/src/expr/mod.rs

```diff
@@ -597,7 +597,30 @@ impl<'a, S: ContextProvider> SqlToRel<'a, S> {
             }
 
             SQLExpr::Struct { values, fields } => {
-                self.parse_struct(values, fields, schema, planner_context)
+                if !fields.is_empty() {
+                    return not_impl_err!("Struct fields are not supported yet");
+                }
+                fn is_named_struct(values: &Vec<sqlparser::ast::Expr>) -> bool {
+                    values
+                        .iter()
+                        .any(|value| matches!(value, SQLExpr::Named { .. }))
+                }
+
+                if is_named_struct(&values) {
+                    return self.create_named_struct(values, schema, planner_context);
```

Comment:
Looks bit untidy but this will be further simplified and addressed as a part of https://github.com/apache/datafusion/issues/11222

Reply:
> Looks bit untidy but this will be further simplified and addressed as a part of #11222

You could introduce StructPlanner for struct and named struct together in a PR

Reply:
Sure. I have made the change.